### DISCUSSION TITLE
/js/index.php removed from OM-20

### DISCUSSION THREAD

**Discussion by addison74** [OpenMage/magento-lts#2284] [Questions & Answers] [answered]
In this PR #1622 the /js/index.php file was removed only from OM-20. I approved this PR from the wrist without further documenting the reasons. If I look in the source code I see that this file has the following description:

```php
/**
 * Proxy script to combine and compress one or few files for JS and CSS
 *
 * Restricts access only to files under current script's folder
 *
```

I came to this file by accident checking what OM displays in the browser when accessing all directory in its root. In 1.9.4..x accessing /js displays ```SYNTAX: index.php/x.js?f=dir1/file1.js,dir2/file2.js``` that normally shouldn't have been displayed.

Was it a good idea to delete this file?

**Comment by justinbeaty** [comment, marked as answer]
It is better these days to just include each script separately. A few comments:

> I agree with this PR, there's no real reason anymore to merge the js via php, with http2 merging is almost completely useless and lighthouse (as far as i know) doesn't suggest merging anymore. so with this PR we remove some server side computation ;-)
[fballiano](https://github.com/fballiano)

> removing it will allow to have whole js/ static with legit "php_flag engine 0" in js/.htaccess and default nginx config serving js/ as static files.
[jouriy](https://github.com/jouriy)

**Comment by addison74** [comment]
Thank you for your comment. I agree with what @fballiano & @youriy say. There is still a small issue the /js directory doesn't have a .htaccess file yet. Should it have one? From a few I see it makes no sense to disable the running of php files since I did not find any in this directory.

**Reply by justinbeaty** [reply]
Yes I think an htaccess would be a great idea if you want to make the PR to the 20.x branch. I guess just:

php_flag engine off

within the two IfModule tags for PHP 7 & 8.

**Reply by addison74** [reply]
I will create a PR for this.
